src/parser/generated_parser.py

```diff
@@ -0,0 +1,7339 @@
+#!/usr/bin/env python3.8
+# @generated by pegen from ../cpython/Grammar/python.gram
+
+import ast
+import sys
+import tokenize
+
+from typing import Any, Optional
+
+from pegen.parser import memoize, memoize_left_rec, logger, Parser
+class GeneratedParser(Parser):
+
+    @memoize
+    def file(self) -> Optional[mod_ty]:
+        # file: statements? $
+        mark = self.mark()
+        cut = False
+        if (
+            (a := self.statements(),)
+            and
```

Comment:
So this is what the equivalent statement is for our || 1 which is clever!
it means that the test is always a tuple of length 1 but a gets assigned the correct thing.

will need to adjust our parser.

(also this is just what the python parser would generate if it was run over the grammar file - one problem that we might face is that the python parser, as far as i know, was only meant to generating the parser to parser the grammar file)

Reply:
Good idea to look at the generated python parser